src/openforms/submissions/api/validation.py

```diff
@@ -78,6 +85,27 @@ def validate_submission_completion(
     completion = InvalidCompletion(
         submission=submission,
         incomplete_steps=incomplete_steps,
+        invalid_prefilled_fields=invalid_prefilled_fields,
         request=request,
     )
     return completion.validate()
+
+
+def get_invalid_prefilled_fields(
+    submission_step: "SubmissionStep", submission: "Submission"
+) -> List[str]:
+    invalid_prefilled_fields = []
+
+    for component in submission_step.form_step.iter_components():
+        if "prefill" not in component or component["prefill"]["plugin"] == "":
+            continue
+
+        plugin_name = component["prefill"]["plugin"]
+        attribute_name = component["prefill"]["attribute"]
+        if (
+            submission.data[component["key"]]
```

Comment:
technically this should be checking `submission_step.data`, but I presume you're doing this on purpose as you can alter data from another step currently? If so, please add a comment to explain this, at some point we'll properly add namespacing which will make this not required anymore.

Reply:
Mhmm I hadn't thought about the fact that other steps could be modified :thinking:  I think it makes sense for now to just check `submission_step.data`